### ISSUE TITLE
Apply the external diff command to the entire directory (opt-in)

### ISSUE THREAD

**Issue by FranklinYu** [twpayne/chezmoi#2828]
## Is your feature request related to a problem? Please describe.

My external diff tool [Difftastic](https://difftastic.wilfred.me.uk) cannot handle directories when being invoked from Chezmoi. Chezmoi seems to use `foobar-directory/null` to indicate that the directory doesn’t exist, which doesn’t work with Difftastic (or as basic as the GNU Diff).

## Describe the solution you'd like

Have another configuration bit under the `diff` section, possibly named `recursive`. If it is true, then Chezmoi only invoke the `diff.command` once, and the two template-variables in `diff.args` becomes the two top-level directories. This works with Difftastic, [Meld](https://en.wikipedia.org/wiki/Meld_(software)), and plain `diff --recursive`.

## Describe alternatives you've considered

An alternative could be to make use of the compatibility of Difftastic against the [`GIT_EXTERNAL_DIFF` API](https://git-scm.com/docs/git#Documentation/git.txt-codeGITEXTERNALDIFFcode). However, I’m not aware of any other diff tool compatible with that specific interface out-of-box; for example, I use Meld with Git under `git difftool` instead of `git diff`, and the `difftool` subcommand provides [`--dir-diff`](https://git-scm.com/docs/git-difftool#Documentation/git-difftool.txt--d) which is similar to what I proposed (comparing the project-root directory).

Another alternative could be to provide `{{ .DestinationDir }}` and `{{ .TargetDir }}`, but Chezmoi still need to decide how many times to invoke the command.

## Additional context

N/A

**Comment by twpayne** [comment]
I haven't tried this, but you might be able to set

```toml
[diff]
    exclude = ["dirs"]
```

in your config file, which should stop chezmoi from invoking your diff tool for directories.

**Comment by halostatue** [comment]
There might be a separate discussion that could be made for different tools on different types. Something like:

```toml
[diff.dirs]
  command = "" # some tool that shows a directory diff
```

**Comment by twpayne** [comment]
> I haven't tried this, but you might be able to set
> 
> ```toml
> [diff]
>     exclude = ["dirs"]
> ```
> 
> in your config file, which should stop chezmoi from invoking your diff tool for directories.

This indeed fixes the problem, as verified in #2830.

**Comment by FranklinYu** [comment]
I’m not sure whether to consider that a “fix”. If I understand it correctly, that configuration would completely suppress any output for directories, so when I run `chezmoi diff` I don’t even know that it is going to create a directory upon `chezmoi apply`. This is *not* what I really meant.
